MetaMask/metamask-extension
# UX: Added Account list pinning and Unpinning

This PR is to add accounts pinning in the account list menu

## **Description**

List management for accounts has been one of the most [requested features](https://docs.google.com/presentation/d/1mpgU2Zixmn6bb8Y3LPMFRUCQxFiEbYwGEWuzNx8jvBo/edit#slide=id.g2588819e87c_6_0) from the community. In this iteration, we'll implement account list management through pin and unpinning accounts.

## **Related issues**

Fixes: #20918 

## **Manual testing steps**

1. Run the extension with `NETWORK_ACCOUNT_DND=1` yarn start command
2. Go to account list menu
3. Click on the options
4. Click on Pin Account to Top
5. See the pinned account is on top
6. Click on Pin Account for a new account, this account should be on second position from top
7. Unpin Accounts

## **Screenshots/Recordings**


### **Before**
![Screenshot 2023-11-17 at 1 11 16 PM](https://github.com/MetaMask/metamask-extension/assets/39872794/0fa9bbe1-c614-44a3-a0a2-14bc0976dc9a)


### **After**



https://github.com/MetaMask/metamask-extension/assets/39872794/abf07bf3-9baf-49d4-b27e-20ed3504d493





## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
